internal/docker/client.go

```diff
@@ -134,32 +135,45 @@ type Cli struct {
 }
 
 func NewDockerClient(ctx context.Context, env Env) Client {
+	var failureReason string
+	var serverVersion types.Version
+	defer func() {
+		reportDockerClientConnect(ctx, failureReason, env, serverVersion)
```

Comment:
i understand what you're trying to do here, but this is going to give you bogus and unreliable metrics.

in particular, lots of people 
- use tilt with pure local services and docker turned off
- use tilt with minikube's docker socket (and no local docker socket)

the reason we have to use the exploding client pattern here is because we don't know until much later whether we're connecting to the right docker socket.

i think this logic probably needs to live in the tiltfile reconciler (see the SetOrchestrator call, which is where we determine which docker socket is the right one to connect to)